supervisor/hardware/data.py

```diff
@@ -0,0 +1,36 @@
+"""Data representation of Hardware."""
+from pathlib import Path
+from typing import Dict, List, Optional
+
+import attr
+
+
+@attr.s(slots=True, frozen=True)
+class Device:
+    """Represent a device."""
+
+    name: str = attr.ib(eq=False)
+    path: Path = attr.ib(eq=False)
+    sysfs: Path = attr.ib(eq=True)
+    subsystem: str = attr.ib(eq=False)
+    links: List[Path] = attr.ib(eq=False)
+    attributes: Dict[str, str] = attr.ib(eq=False)
+
+    @property
+    def cgroups_major(self) -> int:
+        """Return Major cgroups."""
+        return int(self.attributes.get("MAJOR", 0))
+
+    @property
+    def cgroups_minor(self) -> int:
+        """Return Major cgroups."""
+        return int(self.attributes.get("MINOR", 0))
+
+    @property
+    def by_id(self) -> Optional[Path]:
```

Comment:
There are other symlinks than `by-id` (e.g. for partitions/disks knows by-uuid etc...). Not sure if we should consider doing this more generic, e.g. just `device_symlinks` and than make it a list...? But I guess this is internals we can extend rather easily in a second step if necessary.

Reply:
We have this list above. But 99% of the links can't be used by docker because of special chars. Only this link and that is what user search

Reply:
It's is only a helper for UI